scenarioo-client/app/manage/comparisons/createComparisonModal.controller.js

```diff
@@ -0,0 +1,201 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+angular.module('scenarioo.controllers').controller('CreateComparisonModalController', CreateComparisonModalController);
+
+function CreateComparisonModalController($uibModalInstance, BranchesAndBuildsService, BuildDiffInfosResource, ComparisonCreateResource) {
+
+    var vm = this;
+
+    vm.branchesAndBuilds = {};
+    vm.comparisonName = '';
+    vm.baseBranch = null;
+    vm.baseBuild = null;
+    vm.comparisonBranch = null;
+    vm.comparisonBuild = null;
+    vm.validationMessage = null;
+
+    vm.validateComparisonName = validateComparisonName;
+    vm.setBaseBranch = setBaseBranch;
+    vm.setBaseBuild = setBaseBuild;
+    vm.setComparisonBranch = setComparisonBranch;
+    vm.setComparisonBuild = setComparisonBuild;
+
+    vm.cancel = cancel;
+    vm.createComparison = createComparison;
+
+    activate();
+
+    function activate() {
+        BranchesAndBuildsService.getBranchesAndBuilds().then(function onSuccess(branchesAndBuilds) {
+            vm.branchesAndBuilds = branchesAndBuilds;
+            vm.baseBranch = branchesAndBuilds.selectedBranch;
+            vm.baseBuild = branchesAndBuilds.selectedBuild;
+            loadComparisonsOfCurrentBuild();
+        });
+    }
+
+    function loadComparisonsOfCurrentBuild() {
+        var baseBranchName = vm.baseBranch.branch.name;
+        var baseBuildName = vm.baseBuild.linkName;
+        BuildDiffInfosResource.query(
+            {'baseBranchName': baseBranchName, 'baseBuildName': baseBuildName},
+            function onSuccess(buildDiffInfos) {
+                vm.comparisonsOfCurrentBuild = buildDiffInfos;
+                validateDistinctBuilds();
+            }, function onFailure() {
+                vm.comparisonsOfCurrentBuild = [];
+                validateDistinctBuilds();
+            }
+        )
+    }
+
+    function setBaseBranch(branch) {
+        vm.baseBranch = branch;
+        vm.baseBuild = null;
+        vm.comparisonsOfCurrentBuild = [];
+        vm.validationMessage = null;
+    }
+
+    function setBaseBuild(build) {
+        vm.baseBuild = build;
+        validateBaseBuild();
+        loadComparisonsOfCurrentBuild();
+    }
+
+    function setComparisonBranch(branch) {
+        vm.comparisonBranch = branch;
+        vm.comparisonBuild = null;
+        vm.validationMessage = null;
+    }
+
+    function setComparisonBuild(build) {
+        vm.comparisonBuild = build;
+        validateComparisonBuild();
+    }
+
+    function validateComparisonName() {
```

Comment:
Currently it is as follow: validation always validates only the field the user changed the last. When clicking on submit it will revalidate everything again and show any error that occurs. 

Reason for this: keep it simple and do not validate fields that the user did maybe not interact with.

Of course this is far from a luxury validation solution. But it works so far and I do not want to put more effort into that right now.